torch/nn/functional.py

```diff
@@ -4990,6 +4990,266 @@ def multi_head_attention_forward(
           :math:`S` is the source sequence length. If ``average_attn_weights=False``, returns attention weights per
           head of shape :math:`(num_heads, L, S)` when input is unbatched or :math:`(N, num_heads, L, S)`.
     """
+    why_not_fast_path = can_use_fastpath(query,
+                                         key,
+                                         value,
+                                         embed_dim_to_check,
+                                         num_heads,
+                                         in_proj_weight,
+                                         in_proj_bias,
+                                         bias_k,
+                                         bias_v,
+                                         add_zero_attn,
+                                         dropout_p,
+                                         out_proj_weight,
+                                         out_proj_bias,
+                                         training,
+                                         key_padding_mask,
+                                         attn_mask)
+    if not why_not_fast_path:
+        in_proj_weight_ = torch.empty((3 * embed_dim_to_check, embed_dim_to_check), device=query.device, dtype=query.dtype) \
```

Comment:
What are these Tensors with un-initialized memory used for?

Reply:
They are used in torch._native_multi_head_attention(), which permits Nullable in_proj_weight argument.
Additional tensors introduced to avoid interruption with later called _multi_head_attention_forward_impl.